src/server/adminJs/tabs/projectsTab.ts

```diff
@@ -371,6 +373,46 @@ export const updateStatusOfProjects = async (
   };
 };
 
+function convertToCSV(data) {
+  const headers = Object.keys(data[0]);
+  const rows = data.map(row =>
+    headers.map(header => JSON.stringify(row[header] || '')).join(','),
+  );
+  return [headers.join(','), ...rows].join('\n');
+}
+
+export const exportEmails = async (
+  context: AdminJsContextInterface,
+  request: AdminJsRequestInterface,
+) => {
+  const { records, currentAdmin } = context;
+  const projectIds = request?.query?.recordIds
+    ?.split(',')
+    ?.map(strId => Number(strId)) as number[];
+  const projects = await Project.createQueryBuilder('project')
+    .select(['project.title'])
+    .leftJoin('project.adminUser', 'adminUser')
+    .addSelect('adminUser.email')
+    .where('project.id IN (:...ids)', { ids: projectIds })
+    .getMany();
+  const data = projects.map(p => ({
+    title: p.title,
+    email: p.adminUser.email,
+  }));
+  const csv = convertToCSV(data);
+  const filePath = path.join(__dirname, 'exports', 'emails.csv');
+  fs.mkdirSync(path.dirname(filePath), { recursive: true });
+  fs.writeFileSync(filePath, csv);
```

Comment:
Do you consider the scenario that the file exists from before?
(Does it append the data or over write them?)

Reply:
Yes, it will overwrite